### PR TITLE
Write results to an excel file #4

### LINKED ISSUE #4: Generate the excel file
Create module/file with functions that accept files/filepaths, and create an output excel spreadsheet

### PR DESCRIPTION
**Pull request by mariellemanlulu**
# Summary of Changes

* Added `AmrOutput` class to store results file information
   * file name
   * file sheet name
   * file contents 
* Added `amr_downloader` module
   1. grabs all Complete Amr analysis submission available to the specified project
   2. uses analysis submission ids to grab all the result file objects
      * file object is created in `irida_api`
   3. calls `_write_to_excel`
      * converts each file contents to dataframe
      * writes dataframe to excel as a separate sheet 
* Added logging messages
* Rework `irida_api` to provide analysis submissions instead of analysis results.

### CODE REVIEW

File: irida_staramr_results/amr_downloader.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    :param project_id:
+    :param output_file_name:
+    :return:
+    """
+
+    logging.info(f"Requesting completed amr analysis submissions for project id [{project_id}]")
+    amr_completed_analysis_submissions = irida_api.get_amr_analysis_submissions(project_id)
+
+    if len(amr_completed_analysis_submissions) < 1:
+        logging.info(f"No completed amr analysis submission type for project id [{project_id}].")
```
Comment (JeffreyThiessen) [inline irida_staramr_results/amr_downloader.py]: This should likely be a `warning` as nothing is getting written

File: irida_staramr_results/api/irida_api.py [hunk 2]
```diff
@@ -270,34 +271,62 @@ def _get_link(self, target_url, target_key, target_dict=None):
...
-        amr_analysis_results = [analysis_result for analysis_result in all_analysis_results if
-                                self._is_type_amr(analysis_result)]
+        # Filter AMR Detection type
+        amr_analysis_submissions = [analysis_submission for analysis_submission in project_analysis_submissions
+                                    if self._is_submission_type_amr(analysis_submission)]
 
-        return amr_analysis_results
+        return amr_analysis_submissions
 
     def _get_analysis_submissions(self, project_id):
```
Comment (JeffreyThiessen) [inline irida_staramr_results/api/irida_api.py:320]: All these functions are small and easy to understand and read. Great Job!

**Review by JeffreyThiessen — CHANGES_REQUESTED** [review]
Looks great overall, only had a few comments. 

Also, a nice shorthand you can consider while writing your functions is that `a = dict()` is equivalent to `a = {}`, and `b = list()` is equivalent to `b = []`.

File: irida_staramr_results/amr_downloader.py (hunk 1 above)
Comment (mariellemanlulu) [reply irida_staramr_results/amr_downloader.py]: Done

**Review by JeffreyThiessen — CHANGES_REQUESTED** [review]
Everything looks good, but we need to add `xlsxwriter` to the `requirements.txt` file for it to work. Once that's in it will be good to merge.

**Comment by mariellemanlulu** [comment]
> Everything looks good, but we need to add `xlsxwriter` to the `requirements.txt` file for it to work. Once that's in it will be good to merge.

Done

**Pull request merged**